### ISSUE TITLE
Proposal: Use FlowJS and compile to asm.js

### ISSUE THREAD

**Issue by sasha240100** [WhitestormJS/whitestorm.js#121]
As we work with webgl - speed matters. I seriously think now about using Flow and then compile to Asm.js.
**Links**:
 - https://flowtype.org/
 - https://github.com/RReverser/babel-plugin-asm-js

###### Version:
 - [x] r12 (dev)
 - [ ] r11
 - [ ] r10
 - [ ] r9
 - [ ] r9 (dev)
 - [ ] r8
 - [ ] r7

###### Issue type:
 - [ ] Bug
 - [x] Proposal/Enhancement
 - [ ] Question

------

<details> 
<summary> <b>Tested on: </b> </summary>
###### --- Desktop
   - [ ] Chrome
   - [ ] Chrome Canary
   - [ ] Chrome dev-channel
   - [ ] Firefox
   - [ ] Opera
   - [ ] Microsoft IE
   - [ ] Microsoft Edge

###### --- Android
   - [ ] Chrome
   - [ ] Firefox
   - [ ] Opera

###### --- IOS
   - [ ] Chrome
   - [ ] Firefox
   - [ ] Opera

</details>

**Comment by preco21** [comment]
Hmm, IDK.

Is this really worth to improve the performance than refactoring costs?

**Comment by sasha240100** [comment]
@preco21 let's keep it for featured releases

**Comment by thejmazz** [comment]
This looks cool - was not aware there was a flow -> asm tool. However may not provide much performance boost if underlying Three itself is not also written as Flow.

**Comment by sasha240100** [comment]
@thejmazz I agree. Three.js should be affected too. But I think they will soon use SIMD and other optimizations. That would be cool for physics too
